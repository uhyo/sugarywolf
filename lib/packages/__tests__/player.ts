import {
    Game,
} from '../../game';
import players from '../players';

describe('Player module', ()=> {
    let p: any;
    let game: Game;
    beforeEach(()=> {
        game = new Game();
        p = players.make(game);
    });

    it('name', ()=> {
        expect(players.name).toBe('core.players');
    });

    it('remembers added player', ()=> {
        p.add('player1', 'Player');

        expect(p.get('player1')).toMatchSnapshot();
    });

    it('non-added player is null', ()=> {
        p.add('player1', 'Player');

        expect(p.get('foobar')).toBeNull();
    });

    it('iterates all players', ()=> {
        p.add('player1', 'Player');
        p.add('player2', 'Player2');

        const pls = Array.from(p.all());
        expect(pls).toMatchSnapshot();
    });

    describe('die event', ()=> {
        it('kills player for die event', ()=> {
            p.add('player1', 'Player');

            expect(p.get('player1').dead).toBe(false);
            p.die.emit({
                id: 'player1',
                reason: 'foo',
            });
            expect(p.get('player1').dead).toBe(true);
        });
    });
});
