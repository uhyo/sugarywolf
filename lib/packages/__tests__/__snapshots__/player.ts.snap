// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player module iterates all players 1`] = `
Array [
  Array [
    "player1",
    Object {
      "dead": false,
      "id": "player1",
      "name": "Player",
    },
  ],
  Array [
    "player2",
    Object {
      "dead": false,
      "id": "player2",
      "name": "Player2",
    },
  ],
]
`;

exports[`Player module remembers added player 1`] = `
Object {
  "dead": false,
  "id": "player1",
  "name": "Player",
}
`;
